data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -134,4 +134,9 @@ private boolean isAddressDefinedLocally(final String address) {
         }
     }
 
+    @Override
+    public boolean isReadyForShutdown() {
+        return peerForwarderReceiveBuffer.isEmpty();
```

Comment:
Can we move the logic that deals with the buffer drain timeout to this method? That should help reduce the amount of peer-forwarding configuration that has to leak out of here.

Reply:
See comment above